src/js/components/widgets-link/widget-link-number.react.js

```diff
@@ -0,0 +1,287 @@
+        this.range = this.max - this.min;
+        this.overPoint = false;
+        this.prevOffset = 0;
+        this.prevWheelOffset = 0;
+        this.scale = 2;
+        this.center = (this.width / this.scale);
+        this.offsetX = this.value * this.center;
+
+        this.value = parseFloat(this.props.value);
+        this._setValue(this.value);
+
+        this.drag = false;
+
+        this.handleClick = this.handleClick.bind(this);
+        this._setValue = this._setValue.bind(this);
+        this._drawCanvas = this._drawCanvas.bind(this);
+        this.onMouseDown = this.onMouseDown.bind(this);
+        this.onMouseMove = this.onMouseMove.bind(this);
+        this.onMouseUp = this.onMouseUp.bind(this);
+        this.onMouseLeave = this.onMouseLeave.bind(this);
+        this.onWheel = this.onWheel.bind(this);
+    }
+
+    /**
+     * React lifecycle method called once DIV is mounted
+     */
+    componentDidMount () {
+        this._drawCanvas();
+    }
+
+    /**
+     * Draws the canvas
+     */
+    _drawCanvas () {
+        this.ctx = this.refs.canvas.getContext('2d');
+        this.ctx.clearRect(0, 0, this.width, this.height);
+
+        // horizontal line
+        this.ctx.strokeStyle = this.dimColor;
+        this.ctx.lineWidth = 1;
+        this.ctx.beginPath();
+        this.ctx.moveTo(0, 0.5 + this.height * 0.5);
+        this.ctx.lineTo(0 + this.width, 0.5 + this.height * 0.5);
+        this.ctx.closePath();
+        this.ctx.stroke();
+
+        // vertical line
+        this.ctx.strokeStyle = this.fnColor;
+        this.ctx.lineWidth = 1;
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.width * 0.5, 0);
+        this.ctx.lineTo(this.width * 0.5, this.height);
+        this.ctx.closePath();
+        this.ctx.stroke();
+
+        // Triangle line
+        this.ctx.fillStyle = this.overPoint ? this.selColor : this.fnColor;
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.width * 0.5, 5);
+        this.ctx.lineTo(this.width * 0.48, 0);
+        this.ctx.lineTo(this.width * 0.52, 0);
+        this.ctx.closePath();
+        this.ctx.fill();
+
+        let times = 3;
+        let unit = 40;
+        let step = this.width / unit;
+        let sections = unit * times;
+
+        let offsetX = this.offsetX;
+
+        if (Math.abs(this.offsetX - this.width * 0.5) > this.width * 0.5) {
+            offsetX = (this.offsetX - this.width * 0.5) % (this.width * 0.5) + this.width;
+        }
+
+        this.ctx.strokeStyle = this.dimColor;
+        this.ctx.beginPath();
+        for (let i = 0; i < sections; i++) {
+            let l = (i % (unit / 2) === 0) ? this.height * 0.35 : (i % (unit / 4) === 0) ? this.height * 0.2 : this.height * 0.1;
+            this.ctx.moveTo(i * step - offsetX, this.height * 0.5 - l);
+            this.ctx.lineTo(i * step - offsetX, this.height * 0.5 + l);
+        }
+        this.ctx.stroke();
+
+        let val = Math.round(((this.value - this.min) / this.range) * this.width);
```

Comment:
Variables that are not reassigned should use `const` instead of `let`: `times`, `unit`, `step`, `sections`, `l`, and `val` seem to remain constant (but doublecheck)